File: forge/src/main/java/dev/kineticcat/hexportation/forge/transfer/FluidVariantAttributesImpl.java
package dev.kineticcat.hexportation.forge.transfer;

import dev.kineticcat.hexportation.api.transfer.FluidVariantAttributes;
import net.minecraft.network.chat.Component;
import net.minecraftforge.fluids.FluidStack;

/**
 * Forge implementation using singleton injection pattern.
 */
public class FluidVariantAttributesImpl extends FluidVariantAttributes.Implementation {
    
    public static final FluidVariantAttributesImpl INSTANCE = new FluidVariantAttributesImpl();
    
    /**
     * Get the display name for a fluid variant.
     * For Forge, we extract the FluidStack from our FluidVariant wrapper
     * and use its getDisplayName() method.
     */
    @Override
    public Component getName(Object fluidVariant) {
        if (fluidVariant instanceof FluidVariantImpl.FluidStackWrapper wrapper) {
            FluidStack stack = wrapper.getFluidStack();
            return stack.getDisplayName();
        }
        
        // Fallback for unexpected types
        return Component.literal("Unknown Fluid");
    }
}

File: forge/src/main/java/dev/kineticcat/hexportation/forge/transfer/FluidVariantAttributesImpl.kt
package dev.kineticcat.hexportation.forge.transfer

import dev.kineticcat.hexportation.api.transfer.FluidVariantAttributes
import net.minecraft.network.chat.Component
import net.minecraftforge.fluids.FluidStack

/**
 * Forge implementation using singleton injection pattern.
 */
class FluidVariantAttributesImpl : FluidVariantAttributes.Implementation() {
    
    companion object {
        @JvmField
        val INSTANCE = FluidVariantAttributesImpl()
    }
    
    /**
     * Get the display name for a fluid variant.
     * For Forge, we extract the FluidStack from our FluidVariant wrapper
     * and use its getDisplayName() method.
     */
    override fun getName(fluidVariant: Any): Component = when (fluidVariant) {
        is FluidVariantImpl.FluidStackWrapper -> fluidVariant.getFluidStack().displayName
        else -> Component.literal("Unknown Fluid")
    }
}